src/client/components/meal/AddMeal.js

```diff
@@ -0,0 +1,133 @@
+import React, { useEffect, useState } from "react";
+
+function AddMeal({ addMeal }) {
+  const initialstateForMeal = {
+    title: "",
+    description: "",
+    location: "",
+    when: "",
+    max_reservations: "",
+    price: "",
+    created_date: "",
+  };
+  const [meal, setMeal] = useState(initialstateForMeal);
+  const [error, setError] = useState({});
+
+  const handleChange = (event) => {
+    const { name, value } = event.target;
+    setMeal((prevMeal) => {
+      return {
+        ...prevMeal,
+        [name]: value,
+      };
+    });
+  };
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    setError(validate(meal));
+
+    addMeal(meal);
```

Comment:
```suggestion
    const { isValid, errors } = validate(meal);
    
    if (isValid) {
      addMeal(meal);
    } else {
      setError(errors);
    }
```

Your current code does some validation, but even if there is an error you call `addMeal`.